### PR TITLE
[fix][broker] Do not record as a topic load failure for the requested broker is not the owner

### PR DESCRIPTION
**Pull request by codelipenghui**
### Motivation

The topic_load_failed metric is introduced by https://github.com/apache/pulsar/pull/19236 which is used to monitor the topic load failures such as timeout, metadata service issue, etc.

But the topic owner check exception is an expected behavior if the requested broker is not the owner of the topic. We should avoid to record as a topic load failure from topic owner check. 

For the non-persistent topic, we don't need this change. 

https://github.com/apache/pulsar/blob/d759a1e616614ab661c3f858292dcada8f733e63/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java#L1351-L1364

### Verifying this change

- New unit tests were added

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [x] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -1318,8 +1318,9 @@ public void deleteTopicAuthenticationWithRetry(String topic, CompletableFuture<V
     private CompletableFuture<Optional<Topic>> createNonPersistentTopic(String topic) {
         CompletableFuture<Optional<Topic>> topicFuture = new CompletableFuture<>();
         topicFuture.exceptionally(t -> {
+            // If the broker is not the owner of the topic, the client will redirect to another broker.
+            // In this case, we should not record as a topic load failure.
             pulsarStats.recordTopicLoadFailed();
-            pulsar.getExecutor().execute(() -> topics.remove(topic, topicFuture));
```
Comment (lhotari) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1322]: Removing this line doesn't seem to be related to topic load failed metrics. Is this intentional?
It seems that in the original solution this would ensure that failed topics don't added to the topics map.

**Comment by codelipenghui** [comment]
There is a PIP to improve this case https://github.com/apache/pulsar/pull/23351, which is better than just handle the ServiceNotReady exception. I will close this PR.
